lib/appsignal/logger/backend.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Appsignal.Logger.Backend do
+  @behaviour :gen_event
+
+  def init({__MODULE__, group}) do
+    {:ok, group}
+  end
+
+  def init(_) do
+    {:ok, "phoenix"}
+  end
+
+  def handle_event({level, _gl, {Logger, message, _timestamp, metadata}}, group) do
+    Appsignal.Logger.log(level, group, IO.chardata_to_string(message), Enum.into(metadata, %{}))
```

Comment:
Note that, in addition to any user-provided metadata, the message comes with some Elixir-provided metadata: https://hexdocs.pm/logger/1.12.3/Logger.html#module-metadata

It seems to me that some of these will not make it through to AppSignal, because they're composite values (like `crash_reason` or `domain` -- maybe also `mfa`? the docs aren't clear on the format)

It may be interesting to "massage" known ones, or at least those that we consider useful, into strings (probably by calling `inspect()` on them, or maybe something specific to `mfa` so it shows up as `"Module.function/arity"`?)

Reply:
The metadata gets filtered through https://github.com/appsignal/appsignal-elixir/blob/logger-backend/lib/appsignal/utils/data_encoder.ex, and seems to come in either as types we can handle, or inspected as strings. Although we don’t have specific handling, this should be working like we expect.

Reply:
Awesome! I didn't notice that the metadata went through the data encoder. Marking this as resolved then.